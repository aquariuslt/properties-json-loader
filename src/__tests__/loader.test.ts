import * as webpack from 'webpack';
import * as path from 'path';
import * as fs from 'fs';
import compile from './compile';
import execute from './execute';

const readOutputAsset = (stats: webpack.Stats, assetFilename = 'bundle.js') => {
  const outputPath = stats.compilation.outputOptions.path;

  let output = '';

  try {
    output = fs.readFileSync(path.join(outputPath, assetFilename)).toString();
  } catch (error) {
    output = error.toString();
  }

  return output;
};


describe('properties-json-loader', () => {

  it('# should compile options convert properties to flatten json.', async (done) => {
    const stats = await compile('__fixtures__/index.js', {
      esModule: true
    });

    const output = execute(readOutputAsset(stats));


    const EXPECT_OUTPUT = `{"test":{"key":"Test key label"}}`;

    expect(JSON.stringify(output)).toBe(EXPECT_OUTPUT);
    expect(output).toMatchSnapshot();
    done();
  });

  it('# should log error when parse non-well formatted properties file.', async (done) => {

    try {
      await compile('__fixtures__/sample-error.properties');
    } catch (e) {
      expect(e).toMatchSnapshot();
    }

    done();
  }, 20000);
});
