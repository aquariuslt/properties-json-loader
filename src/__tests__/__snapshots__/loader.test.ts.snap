// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`properties-json-loader # should compile options convert properties to flatten json. 1`] = `
Object {
  "test": Object {
    "key": "Test key label",
  },
}
`;

exports[`properties-json-loader # should default options convert properties to flatten json. 1`] = `"module.exports = {\\"test\\":{\\"key\\":\\"Test key label\\"}}"`;

exports[`properties-json-loader # should log error when parse non-well formatted properties file. 1`] = `null`;
